### ISSUE TITLE
New Events: onTimeLimitChange(newLimit:int, byPlayer: PlayerObject) and onScoreLimitChange(newLimit:int, byPlayer: PlayerObject)

### ISSUE THREAD

**Issue by AndreaCasaluci** [haxball/haxball-issues#2087]
Hello Basro,
I would like to request two new function for the Headless API which is triggered when TimeLimit and ScoreLimit change.
This is similar to my [previous request](https://github.com/haxball/haxball-issues/issues/2077) that was implemented, so I hope this won't be that hard.
Those events could help prevent cases where a normal user finds himself being admin in an automated room and, by setting the TimeLimit and ScoreLimit to some random values, would bring the room to chaos.

Example:
```
room.onTimeLimitChange = function(newLimit:int, byPlayer: PlayerObject) {
if(byPlayer){
 // Check if byPlayer is an actual room admin and stuff
}
else room.setTimeLimit(defaultTimeLimitValue);
});

room.onScoreLimitChange = function(newLimit:int, byPlayer: PlayerObject) {
if(byPlayer){
 // Check if byPlayer is an actual room admin and stuff
}
else room.setScoreLimit(defaultScoreLimitValue);
});
```

**Comment by kadavrahaxball** [comment]
dont wait for basro's update. it aint gonna happen. learn to create custom events for yourself. its possible.

**Comment by Anooxy** [comment]
??
Only admins can change time and score limits... Another unnecessary things..

**Comment by AndreaCasaluci** [comment]
> ?? Only admins can change time and score limits... Another unnecessary things..

@Anooxy did u check the issue I've mentioned in this current post?
It was a similar request and in that post I've explained why it was useful.
If that request was fulfilled in a few hours, I don't understand why this one should be considered useless as well.
I can't see what's your problem with a simple request guys, really.

**Comment by Anooxy** [comment]
If u really need something like this in your script - your script is not good then. Save some "headadmins auth in external db or even in script"...

**Comment by AndreaCasaluci** [comment]
> If u really need something like this in your script - your script is not good then. Save some "headadmins auth in external db or even in script"...

@Anooxy First of all, I never said that my script is good. Even if it was perfect, sometimes it happens that there are no admins (the real ones, saved in MongoDB) and the last one to log off leaves temporary admins for someone in the room in case someone with powers is needed. It has become a habit that we implement, but once (once in months) it happened that we left the powers to the wrong person who set the Score Limit and Time Limit values differently.
So, if there was a possibility to include the event (like so many other events already) that triggers when these values are changed, what is the need to go against this simple request?
I still don't understand what your problem is guys.

**Comment by Anooxy** [comment]
Buuut, why dont u check timeLimit and scoreLimit in onGameStart event?
Just check 
`
if(room.getScores().timeLimit != 5 {
    room.setTimeLimit(5);
}
`

**Comment by AndreaCasaluci** [comment]
> Buuut, why dont u check timeLimit and scoreLimit in onGameStart event? Just check `if(room.getScores().timeLimit != 5) room.setTimeLimit(5);` of course room.stopGame room.setTimeLimit and then room.startGame...

@Anooxy Yes, this would definitely work, but I personally requested the events described in the post because they would allow the user who did this to be traced and having an automated warning script, it would be possible to send a direct warning to this user with these events.
The warning script works with DB and sends the Warning Message to the user if he's in room, or whenever the user enters again (even with a clone). When the user gets an X number of warnings he'll get banned.
The main reason is this.

**Comment by jerryoldson** [comment]
there's an npm package called "node-haxball" and it has what you want. you should take a look if you want
